lib/src/native/realm_core.dart

```diff
@@ -225,7 +225,9 @@ class _RealmCore {
       } else if (config is DisconnectedSyncConfiguration) {
         _realmLib.realm_config_set_force_sync_history(configPtr, true);
       }
-
+      if (config.encryptionKey != null && config is! InMemoryConfiguration) {
```

Comment:
How can we set `config.encryptionKey` on an `InMemoryConfiguration`? I'd rather have this be an assert rather than silently ignoring it.